src/java/main/org/apache/zookeeper/server/FinalRequestProcessor.java

```diff
@@ -476,6 +624,33 @@ private boolean connClosedByClient(Request request) {
         return request.cnxn == null;
     }
 
+    private String getHostAddress(ServerCnxn cnxn) {
+        try {
+            if (cnxn == null) {
+                return "";
+            }
+            InetSocketAddress remoteSocketAddress = cnxn
+                    .getRemoteSocketAddress();
+            if (remoteSocketAddress == null) {
+                return "";
+            }
+            InetAddress address = remoteSocketAddress.getAddress();
+            if (address == null) {
+                return "";
+            }
+            return address.getHostAddress();
+        } catch (Exception e) {
+            // ignore
+        }
+        return "";
+    }
+
+    private String getSessionId(ServerCnxn cnxn) {
```

Comment:
perhaps we can move this string formatting code to somewhere more appropriate

Reply:
moved to ServerCnxn